### PR TITLE
4 issues Add generated during compilation "version.h" file that will contain Kuzya version and build

### PR DESCRIPTION
**Pull request by dmykuliak-cat**

### CODE REVIEW

File: src/kuzya.cpp [hunk 1]
```diff
@@ -1027,9 +1027,17 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
 **/
 void Kuzya::slotAbout(void)
 {
+
+//set build date and version
+    KuziaVersion kuziaversion;
```
Comment (alexchmykhalo) [inline src/kuzya.cpp]: Please, use QVersionNumber instead: http://doc.qt.io/qt-5/qversionnumber.html

File: src/kuzya.cpp [hunk 2]
```diff
@@ -1027,9 +1027,17 @@ void Kuzya::paintWarningMarkers(QList<Compiler::compilerWarning>* warningList)
...
+    BuildTime buildtime;
```
Comment (alexchmykhalo) [inline src/kuzya.cpp]: Please, use QDate, QTime, QDateTime classes to represent date, time and date with time in Qt programs.

File: src/version.h [hunk 3]
```diff
@@ -1,12 +1,139 @@
...
 #define VERSION_H
 
-#define MAJORVER 2
-#define MINORVER 1
-#define REVISION 12
-#define MINOR_REVISION 0
-#define BETA 0
-#define BUILD_TIME "15.02.2012"
-#define KUZYA_VERSION "2.1.12"
+#include <QString>
```
Comment (alexchmykhalo) [inline src/version.h]: Keeping version and date in a separate text file is not a good idea:
 1) it is not a common practice as I know
 2) it is hard to understand meaning of each number (you need to remeber sequence of each in file), it is easy to make a mistake

So, it was ok, that this file contained defines. (See for example, answer here - http://stackoverflow.com/questions/27395120/correct-way-to-encode-embed-version-number-in-program-code) The idea is that all content of this file should be generated by Python script. So, I suggest:
 1) Remove this file from Git completely.
 2) Your python script should generate this file. (Obviously, when you want to build Kuzya, you need to run script with the parameters to create this file, or Kuzya will not compile)

Comment (alexchmykhalo) [reply src/version.h]: If  you implement generation of this file, then  src/BUILD.DAT is not needed.

**Comment by dmykuliak-cat** [comment]
I took into account all your advice and correct my mistakes. But I can not finish it until @OlyaOlga  did not finished #5 .

**Comment by dmykuliak-cat** [comment]
And, as I understand it, I have to add a call python script that building "version.h" in configuration files to the build server.

**Comment by alexchmykhalo** [comment]
Yes, this would be great.

**Comment by dmykuliak-cat** [comment]
@alexchmykhalo all work and build correctly, check it, if you have remark say.

**Comment by alexchmykhalo** [comment]
Nice!) Please, remove src/version.h from sources and task is ready to merge. I've approved request.

**Review by alexchmykhalo — APPROVED** [review]

**Pull request merged**